src/utils/fetch.js

```diff
@@ -37,6 +38,10 @@ module.exports = (url, options = {}) => {
       })
       .on("error", err => reject(err));
 
+      req.setTimeout((opts.timeout), () => {
```

Comment:
What happens on `req.abort()`?
You should also refactor `Promise.race`

Reply:
there is 2 kind of timeouts on the native http module
1. options.timeout = pre handshake timeout
2. req.setTimeout = request flight timeout

if connection esatblished, req.timeout is ignored and req.setTimeout is required, without it the requests can connection will stay open